gulpfile.js

```diff
@@ -0,0 +1,872 @@
+var minifyShaders = taskName === 'minify' || taskName === 'minifyRelease' || taskName === 'release' || taskName === 'makeZipFile';
+
+var version = packageJson.version;
+if (/\.0$/.test(version)) {
+    version = version.substring(0, version.length - 2);
+}
+
+var sourceFiles = ['Source/**/*.js',
+                   '!Source/*.js',
+                   '!Source/Workers/**',
+                   '!Source/ThirdParty/Workers/**',
+                   'Source/Workers/createTaskProcessorWorker.js'];
+
+var buildFiles = ['Specs/**/*.js',
+                  '!Specs/SpecList.js',
+                  'Source/Shaders/**/*.glsl'];
+
+var jsHintFiles = ['Source/**/*.js',
+                   '!Source/Shaders/**',
+                   '!Source/ThirdParty/**',
+                   '!Source/Workers/cesiumWorkerBootstrapper.js',
+                   'Apps/**/*.js',
+                   'Apps/Sandcastle/gallery/*.html',
+                   '!Apps/Sandcastle/ThirdParty/**',
+                   'Specs/**/*.js',
+                   'Tools/buildTasks/**/*.js',
+                   'gulpfile.js'];
+
+var filesToClean = ['Source/Cesium.js',
+                    'Build',
+                    'Instrumented',
+                    'Source/Shaders/**.js',
+                    'Specs/SpecList.js',
+                    'Apps/Sandcastle/.jshintrc',
+                    'Apps/Sandcastle/jsHintOptions.js',
+                    'Apps/Sandcastle/gallery/gallery-index.js',
+                    'Cesium-*.zip'];
+
+var filesToSortRequires = ['Source/**/*.js',
+                           '!Source/Shaders/**',
+                           '!Source/ThirdParty/**',
+                           '!Source/Workers/cesiumWorkerBootstrapper.js',
+                           '!Source/copyrightHeader.js',
+                           '!Source/Workers/transferTypedArrayTest.js',
+                           'Apps/**/*.js',
+                           '!Apps/Sandcastle/ThirdParty/**',
+                           '!Apps/Sandcastle/jsHintOptions.js',
+                           'Specs/**/*.js',
+                           '!Specs/spec-main.js',
+                           '!Specs/SpecRunner.js',
+                           '!Specs/SpecList.js',
+                           '!Apps/Sandcastle/Sandcastle-client.js',
+                           '!Apps/Sandcastle/Sandcastle-header.js',
+                           '!Apps/Sandcastle/Sandcastle-warn.js',
+                           '!Apps/Sandcastle/gallery/gallery-index.js'];
+
+gulp.task('default', ['combine']);
+
+gulp.task('build', function(done) {
+    mkdirp.sync('Build');
+    glslToJavaScript(minifyShaders, 'Build/minifyShaders.state');
+    createCesiumJs();
+    createSpecList();
+    createGalleryList();
+    createSandcastleJsHintOptions();
+    done();
+});
+
+gulp.task('build-watch', function() {
+    gulp.watch(buildFiles, ['build']);
+});
+
+gulp.task('buildApps', ['combine', 'minifyRelease'], function() {
```

Comment:
So the issue now is that the .js shader files shipped in the zip file are now the minified versions.  It turns out my previous explanation of the old behavior was wrong.  In the Ant build, we minified first, then combined unminified, then built CesiumViewer, which meant that the files in the Source tree in the zip file were the unminified ones, but the built CesiumViewer also used the unminified, which was a bug.

This is all obviously insane.  To avoid space madness, we should change things so that there is always a clear dependency graph from src to dest, and avoid situations like this where two different processes are writing different content to the same dest.  The one exception to this is the contents of the `Build/Cesium` directory which always reflect the last built configuration.  

I have some thoughts on how to improve this, so I think for now it would probably suffice to just hack `makeZipFile` to call `glslToJavaScript(true)` to get the right content in the zip, and then I will work on further improvements as a separate pull request.